components/style/properties/properties.mako.rs

```diff
@@ -588,18 +588,24 @@ pub fn parse_property_declaration_list(context: &ParserContext, input: &mut Pars
             }
         }
     }
-    PropertyDeclarationBlock {
-        declarations: Arc::new(deduplicate_property_declarations(declarations)),
+    let (declarations, removed_any) = deduplicate_property_declarations(declarations);
+    let mut block = PropertyDeclarationBlock {
+        declarations: Arc::new(declarations),
         any_normal: any_normal,
         any_important: any_important,
+    };
+    if removed_any {
+        block.recalc_any();
     }
+    block
 }
 
 
 /// Only keep the "winning" declaration for any given property, by importance then source order.
 /// The input and output are in source order
 fn deduplicate_property_declarations(declarations: Vec<(PropertyDeclaration, Importance)>)
-                                     -> Vec<(PropertyDeclaration, Importance)> {
+                                     -> (Vec<(PropertyDeclaration, Importance)>, bool) {
```

Comment:
hmm... Maybe this would be clearer just comparing the input with the output length? Also, note that we don't need to recalc everything, we can be a bit smarter. If there was any important declaration, `any_important` would still always be true. And if there wasn't any important one, and `any_normal` was true, `any_normal` still needs to be true.